src/styles/main.scss

```diff
@@ -1,3 +1,82 @@
+@use "./variables" as val;
+@use "sass:map";
+
 body {
   margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  position: fixed;
+  width: 780px;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+
+  &__day {
+    background-color: #eee;
+    width: 100px;
+    height: 100px;
+    border: 1px solid black;
+    font-size: 30px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transition: cursor, background-color, transform, 0.5s;
+  }
+
+  &__day:hover {
+    cursor: pointer;
+    background-color: pink;
+    transform: translateY(-20px);
+  }
+  
+  @for $numb-day from 1 through 31 {
+    .calendar__day:nth-child(#{$numb-day})::before {
+      content: "#{$numb-day}";
+    }
+  }
+
+  &--month-length--28 {
+    @for $day from 28 + 1 through 32 {
+      .calendar__day:nth-child(#{$day}) {
+        display: none;
+      }
+    }
+  }
+
+  &--month-length--29 {
+    @for $day from 29 + 1 through 32 {
+      .calendar__day:nth-child(#{$day}) {
+        display: none;
+      }
+    }
+  }
+
+  &--month-length--30 {
+    @for $day from 30 + 1 through 32 {
+      .calendar__day:nth-child(#{$day}) {
+        display: none;
+      }
+    }
+  }
+
+  &--month-length--31 {
+    @for $day from 31 + 1 through 32 {
+      .calendar__day:nth-child(#{$day}) {
+        display: none;
+      }
+    }
+  }
+
```

Comment:
write the loop for these selectors, and feel free to ask for some help in the chat